src/slider.tsx

```diff
@@ -1,164 +1,428 @@
+        const finalValue = value ?? currentValue;
 
-    React.useEffect(() => {
-        if (Array.isArray(steps)) {
-            setMaxSlider(steps.length - 1);
-            let valueIndex = 0;
+        const prevValueRef = React.useRef(finalValue);
 
-            if (value !== undefined) {
-                valueIndex = steps.indexOf(getApproximation(value));
+        const handleChange = (value: number, isPercentage: boolean) => {
+            const realValue = getValueInRange(isPercentage, min, max, step, value);
+            if (prevValueRef.current !== realValue) {
+                onChangeValue(values ? values[realValue] : realValue);
             }
+            setCurrentValue(realValue);
+            prevValueRef.current = realValue;
+        };
+
+        const trackRef = React.useRef<HTMLDivElement>(null);
+        const thumbRef = React.useRef<HTMLDivElement>(null);
+        const sliderRef = React.useRef<HTMLDivElement>(null);
+
+        const [isPointerDown, setIsPointerDown] = React.useState(false);
+        const [isThumbHovered, setIsThumbHovered] = React.useState(false);
+        const [isFocused, setIsFocused] = React.useState(false);
+        const {isIos} = useTheme();
+
+        const isPointerOverElement = (element: HTMLElement | null, x: number, y: number) => {
+            const box = element?.getBoundingClientRect();
+            return box !== undefined && box.left <= x && x <= box.right && box.top <= y && y <= box.bottom;
+        };
+
+        const isTabKeyDownRef = React.useRef(false);
 
-            setValueRanger(valueIndex);
-        } else {
-            setMinSlider(min);
-            setMaxSlider(max);
-            let ranger = min;
-            if (value !== undefined) {
-                if (min > value || value > max) {
-                    ranger = max < value ? max : min;
+        React.useEffect(() => {
+            const handleKeyDown = (e: KeyboardEvent) => {
+                switch (e.keyCode) {
+                    case TAB:
+                        isTabKeyDownRef.current = true;
+                        break;
+                    case ESC:
+                        if (isFocused) {
+                            setIsFocused(false);
+                            thumbRef.current?.blur();
+                        }
+                        break;
+                    default:
+                    // do nothing
                 }
+            };
+
+            const handleKeyUp = () => (isTabKeyDownRef.current = false);
+
+            document.addEventListener('keydown', handleKeyDown, false);
+            document.addEventListener('keyup', handleKeyUp, false);
+            return () => {
+                document.removeEventListener('keydown', handleKeyDown, false);
+                document.removeEventListener('keyup', handleKeyUp, false);
+            };
+        }, [isFocused]);
+
+        const handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>) => {
+            switch (e.keyCode) {
```

Comment:
`keyCode` is deprecated, consider using `key`

Reply:
we should remove the key-code constants file (in a different task)

Reply:
we are using keyCode in several components. I believe this is something that should be updated in all the components at once in a separate PR in order to maintain a clean code.

I've created an issue for this: https://jira.tid.es/browse/WEB-1653